src/index.ts

```diff
@@ -81,7 +81,7 @@ type TokenType =
 interface LexToken {
   type: TokenType;
   index: number;
-  value: string;
+  value: string | { name: string; pattern?: string };
```

Comment:
Instead of this, it'd be simpler to add `PATTERN` to `TokenType`. The `value` of the pattern would be what is between the two `()`.